From a8c69585eb2385a1ee4ddd7fe01ac467cbe806b9 Mon Sep 17 00:00:00 2001
From: Christian Ege <k4230r6@gmail.com>
Date: Sun, 30 Oct 2016 05:42:40 +0100
Subject: [PATCH 1/2] Removed dts folder from device tree file name

This is due to the fact that the meta-fsl-arm layer does provide a
SD-card image class which do not put the dtb files in a separate
sub-folder

Signed-off-by: Christian Ege <k4230r6@gmail.com>
---
 board/udoo/neo/neo.c        | 9 ++++-----
 board/udoo/quad_dual/udoo.c | 8 ++++----
 2 files changed, 8 insertions(+), 9 deletions(-)

diff --git a/board/udoo/neo/neo.c b/board/udoo/neo/neo.c
index e2bf13c..369db6b 100644
--- a/board/udoo/neo/neo.c
+++ b/board/udoo/neo/neo.c
@@ -1094,18 +1094,17 @@ int do_udooinit(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 		}
 	}

-	char* dir_part = "dts";
+	char fdt_file[100];
 	char* customdtb = getenv("use_custom_dtb");
+	sprintf(fdt_file, "/boot/%s%s%s.dtb", modelfdt, video_part, m4_part);
 	if (customdtb) {
 		customdtb = trim(customdtb);
 		if (strcmp(customdtb, "true") == 0 || strcmp(customdtb, "yes") == 0 || strcmp(customdtb, "enabled") == 0) {
-			dir_part = "dts-overlay";
+			char* dir_part = "dts-overlay";
+			sprintf(fdt_file, "/boot/%s/%s%s%s.dtb", dir_part, modelfdt, video_part, m4_part);
 		}
 	}

-	char fdt_file[100];
-	sprintf(fdt_file, "/boot/%s/%s%s%s.dtb", dir_part, modelfdt, video_part, m4_part);
-
 	printf("Device Tree: %s\n", fdt_file);
 	setenv("fdt_file", fdt_file);

diff --git a/board/udoo/quad_dual/udoo.c b/board/udoo/quad_dual/udoo.c
index 8abc553..5087f2d 100644
--- a/board/udoo/quad_dual/udoo.c
+++ b/board/udoo/quad_dual/udoo.c
@@ -378,17 +378,17 @@ int do_udooinit(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 		}
 	}

-	char* dir_part = "dts";
+	char fdt_file[100];
 	char* customdtb = getenv("use_custom_dtb");
+	sprintf(fdt_file, "/boot/%s%s.dtb", modelfdt, video_part);
 	if (customdtb) {
 		customdtb = trim(customdtb);
 		if (strcmp(customdtb, "true") == 0 || strcmp(customdtb, "yes") == 0 || strcmp(customdtb, "enabled") == 0) {
-			dir_part = "dts-overlay";
+			char* dir_part = "dts-overlay";
+			sprintf(fdt_file, "/boot/%s/%s%s.dtb", dir_part, modelfdt, video_part);
 		}
 	}

-	char fdt_file[100];
-	sprintf(fdt_file, "/boot/%s/%s%s.dtb", dir_part, modelfdt, video_part);

 	printf("Device Tree: %s\n", fdt_file);
 	setenv("fdt_file", fdt_file);
--
2.7.4

